crates/apollo-schema-index/src/traverse.rs

```diff
@@ -53,67 +53,65 @@ impl SchemaExt for Schema {
-                                        let field_type = field.ty.inner_named_type();
-                                        let field_args = field
-                                            .arguments
-                                            .iter()
-                                            .map(|arg| arg.ty.inner_named_type().clone())
-                                            .collect::<Vec<_>>();
+                if let Some(extended_type) = self.types.get(named_type)
+                    && !extended_type.is_built_in()
+                {
+                    if traverse_children {
+                        match extended_type {
+                            ExtendedType::Object(obj) => {
+                                stack.extend(obj.fields.values().map(|field| {
+                                    let field_type = field.ty.inner_named_type();
+                                    let field_args = field
+                                        .arguments
+                                        .iter()
+                                        .map(|arg| arg.ty.inner_named_type().clone())
+                                        .collect::<Vec<_>>();
+                                    (
+                                        field_type,
+                                        current_path.clone().add_child(
+                                            Some(field.name.clone()),
+                                            field_args,
+                                            field_type.clone(),
+                                        ),
+                                    )
+                                }));
+                            }
+                            ExtendedType::Interface(interface) => {
+                                stack.extend(interface.fields.values().map(|field| {
+                                    let field_type = field.ty.inner_named_type();
+                                    let field_args = field
+                                        .arguments
+                                        .iter()
+                                        .map(|arg| arg.ty.inner_named_type().clone())
+                                        .collect::<Vec<_>>();
+                                    (
+                                        field_type,
+                                        current_path.clone().add_child(
+                                            Some(field.name.clone()),
+                                            field_args,
+                                            field_type.clone(),
+                                        ),
+                                    )
+                                }));
+                            }
+                            ExtendedType::Union(union) => {
+                                stack.extend(union.members.iter().map(|member| &member.name).map(
+                                    |next_type| {
```

Comment:
I’m not sure I see how this change relates to the rust-toolchain or Renovate updates. Is this addressing something else?